lib/disc.rb

```diff
@@ -1,12 +1,18 @@
 require 'date'
 require 'disque'
 require 'msgpack'
+require 'connection_pool'
 
 require_relative 'disc/version'
 
 class Disc
-  attr_reader :disque,
-    :disque_timeout
+  def self.pool
+    @pool ||= begin
+      concurrency = Integer(ENV.fetch('DISC_CONCURRENCY', 25))
+      tolerance = Integer(ENV.fetch('DISC_TOLERANCE', disque_timeout / 1000.0))
```

Comment:
The default timeout for each thread would be the same as for disque to consider a job was not finished. Maybe this should be lower? Dunno, seemed like a decent default.